internal/core/metadata/rest_device.go

```diff
@@ -523,10 +522,11 @@ func restCheckForDevice(w http.ResponseWriter, r *http.Request) {
 			LoggingClient.Debug(fmt.Sprintf("device %s %v", token, err))
 		}
 	}
+
 	//If lookup by name failed, see if we were passed the ID
 	if len(dev.Name) == 0 {
 		if bson.IsObjectIdHex(token) {
```

Comment:
Don't we need to handle the case where it's a UUID as well as a bson.ObjectId here?

Reply:
Actually that check should be inside of dbClient.GetDeviceById. The controller shouldn't do this at all. The validation responsibility is farther down, so this bson check should be removed in my view (provided it's in the dbClient implementation)

Reply:
Ah, right.  I thought this was `pkg/db/mongo/metadata.go` for some reason.  That makes more sense.

Reply:
I concur with Trevor.  It's actually already being handled inside the Mongo implementation of GetDeviceById.  The check for `bson.IsObjectIdHex(token)` has been removed.